fido/Crypto/Fido2/Protocol.hs

```diff
@@ -380,6 +385,35 @@ instance Aeson.ToJSON URLEncodedBase64 where
 data PublicKey = Ec2Key ECDSA.PublicKey
   deriving (Show)
 
+publicKeyPoint :: PublicKey -> ECDSA.Point
+publicKeyPoint (Ec2Key key) =
+  let ECDSA.PublicKey {public_q} = key
+   in public_q
+
+publicKeyX :: PublicKey -> Integer
+publicKeyX key =
+  case publicKeyPoint key of
+    ECDSA.Point x _y -> x
+    -- TODO: Does this ever happen?
```

Comment:
From the COSE docs:

> The encoding of the infinity point is not
        supported.


so no

Reply:
We now have a total injection from our own `Point` to `ECDSA.Point`